cli/azd/pkg/ext/models_test.go

```diff
@@ -0,0 +1,110 @@
+package ext
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_HooksFromFolderPath(t *testing.T) {
+	t.Run("HooksFileExistsYaml", func(t *testing.T) {
+		tempDir := t.TempDir()
+		hooksPath := filepath.Join(tempDir, "azd.hooks.yaml")
+		hooksContent := []byte(`
+pre-build:
+  shell: sh
+  run: ./pre-build.sh
+post-build:
+  shell: pwsh
+  run: ./post-build.ps1
```

Comment:
Nit: The hook configuration doesn't care what the name is but we should use the well known names like `prebuild` or `postbuild` instead of using the hyphens like in these examples just to enforce the consistency with the real hook names.